app/services/encryption/kms_client.rb

```diff
@@ -26,21 +26,46 @@ def use_kms?(ciphertext)
       FeatureManagement.use_kms? && self.class.looks_like_kms?(ciphertext)
     end
 
+    # chunk plaintext into ~4096 byte chunks, but not less than 1024 bytes in a chunk if chunking
```

Comment:
I think the bit about chunks less than 1024 bytes can go in the `encrypt_in_chunks` method so it's a bit closer to the tricky code there.